databases/week2/homework.sql

```diff
@@ -0,0 +1,49 @@
+-- Get all the tasks assigned to users whose email ends in @spotify.com
+SELECT DISTINCT 
+	t.title
+FROM
+	task t
+INNER JOIN user_task ut ON
+	t.id IN (
+	SELECT
+		ut.task_id
+	FROM
+		user_task ut
+	INNER JOIN user u ON
+		ut.user_id = u.id
+	WHERE
+		u.email LIKE '%@spotify.com');
+-- Get all the tasks for 'Donald Duck' with status 'Not started'
+	SELECT
+	t.title
+FROM
+	task t
+INNER JOIN user_task ut ON
+	t.id = ut.task_id
+INNER JOIN user u ON
+	u.id = ut.user_id
+INNER JOIN status s ON
+	s.id = t.status_id
+WHERE
```

Comment:
I see you are performing the same join 3 times over in 3 queries. It is recommended you create a view. In short, views allow you to query data from one or more tables as if it were a single table. They can help simplify queries, hide data complexity, and enhance security.

Create the view using the CREATE VIEW statement:

```sql
CREATE VIEW view_name AS
SELECT column1, column2,... 
FROM table_name
WHERE condition;
```

The view will now exist in your database and you can query it like a normal table:

```sql
SELECT * FROM view_name; 
```

To alter an existing view, use the ALTER VIEW statement:

```sql 
ALTER VIEW view_name AS 
SELECT column1, column2,...
FROM table_name
WHERE condition;
```

To drop a view, use the DROP VIEW statement:

```sql
DROP VIEW view_name;
```

So in your case you can do this:

```sql
-- Create the view 
CREATE VIEW users_tasks AS 
SELECT * FROM task t
INNER JOIN user_task ut ON
	t.id = ut.task_id  
INNER JOIN user u ON
   ut.user_id = u.id;

-- Query the view  
SELECT DISTINCT title 
FROM users_tasks
WHERE email LIKE '%@spotify.com';

```